src/authentication/authentication-hapi-plugin.ts

```diff
@@ -106,9 +94,37 @@ class AuthenticationHapiPlugin extends HapiPlugin {
 
   public async getTeamTokenHandler(request: Hapi.Request, reply: Hapi.IReply) {
     try {
-      const teamId = parseInt(request.paramsArray[0], 10);
-      const teamToken: TeamToken[] = await teamTokenQuery(this.db, undefined, teamId);
-      if (!teamToken.length) {
+      const credentials = request.auth.credentials as AccessToken;
+      const userName = sanitizeUsername(credentials.sub);
+      const isAdmin = await this.isAdmin(userName);
+      const userTeams = await this.gitlab.getUserGroups(userName);
+      const parameterKey = 'teamIdOrName';
+      const teamIdOrName = request.params[parameterKey];
+      let ownTeam: Group | undefined;
+      if (userTeams && teamIdOrName) {
+        ownTeam = userTeams.find(assertTeam.bind(undefined, teamIdOrName));
```

Comment:
I would do this instead of bind

```
      if (userTeams && teamIdOrName) {
        ownTeam = userTeams.find((group => assertTeam(teamIdOrName, group)));
      }
```

This way we don't lose type safety, and would be much more readable for me.